ext/search_kit/ang/crmSearchDisplay/traits/searchDisplayBaseTrait.service.js

```diff
@@ -27,11 +27,8 @@
           this.placeholders.push({});
         }
 
-        this.getResults = _.debounce(function() {
-          $scope.$apply(function() {
-            ctrl.runSearch();
-          });
-        }, 800);
+        // _.debounce used here to trigger the initial search immediately but prevent subsequent launches within 800ms
+        this.getResults = _.debounce(ctrl.runSearch, 800, {leading: true, trailing: false});
```

Comment:
`$scope.$apply` was there for a reason: the debounce function is vanilla javascript so executes outside Angular's scope, so it has to be applied.

Reply:
Just read your explanation, and you may be right that the leading edge stays within the scope because it doesn't dispatch a timeout... but what about the trailing edge when the user is typing and this gets debounced?

Reply:
A way around this might be to wrap `ctrl.runSearch` in an AngularJS `$timeout()` with no delay - that would ensure it always goes through the same path and is always in-scope regardless of leading/trailing edge.

Reply:
Ok sounds like this PR can stay as-is but needs a few people to `r-run` it.